smack-experimental/src/main/java/org/jivesoftware/smackx/mam/provider/MamPrefsIQProvider.java

```diff
@@ -43,6 +45,12 @@
 
     @Override
     public MamPrefsIQ parse(XmlPullParser parser, int initialDepth, XmlEnvironment xmlEnvironment) throws XmlPullParserException, IOException {
+        MamElementFactory elementFactory = null;
+        try {
+            elementFactory = MamVersion.fromNamespace(parser.getNamespace()).newElementFactory();
+        } catch (IllegalArgumentException e) {
+            // do nothing, try to get the namespace from the parser later
```

Comment:
Can you elaborate why something in this `try` block would throw an IAE? I would have expected that `XmlPullParser.getNamespace()` at the starte of a `Provider`'s `parse()` method always returns a namespace. If so, then this should only throw unexpected exceptions (which we usually don't want to catch).

Also please consider only having those parts in the `try` block whos exceptions you intent to throw. For example: Can `parser.getNamespace()` be moved out of this `try` block, because we don't expect to throw it an IAE? What about `newElementFactory`?

Reply:
When running the `MamPrefIQProviderTest` unit test, the initial `parser.getNamespace()` returns `null`, this causes the IAE to be thrown by `MamVersion.fromNamespace()`.